File: private/inet/xml/core/util/namespacecontext.cxx
/*
 * @(#)NameSpaceContext.cxx 1.0 6/10/97
 * 
* Copyright (c) 1997 - 1999 Microsoft Corporation. All rights reserved. * 
 */ 
#include "core.hxx"
#pragma hdrstop

DEFINE_CLASS_MEMBERS(NameSpaceContext, _T("NameSpaceContext"), Base);

NameSpaceContext::NameSpaceContext()
{
    current = Hashtable::newHashtable();
    contexts = Stack::newStack();
}

/**
 *  add name space. long name is the key
 */
void NameSpaceContext::addNameSpace(Atom * url, Atom * n)
{
    current->put(url, n);
}

/**
 * find name space, long name is the key
 */
Atom * NameSpaceContext::findNameSpace(Atom * n)
{
    return (Atom *)current->get(n);
}

void NameSpaceContext::push()
{
    contexts->push(current);
    current = (Hashtable *)current->clone();
}

void NameSpaceContext::pop()
{
    current = (Hashtable *)contexts->pop();
}

File: private/inet/xml/java/com/ms/xml/util/namespacecontext.java
/*
 * @(#)XMLOutputStream.java 1.0 6/10/97
 * 
 * Copyright (c) 1997 Microsoft, Corp. All Rights Reserved.
 * 
 */ 
package com.ms.xml.util;

import java.util.Hashtable;
import java.util.Stack;

public class NameSpaceContext
{
	/**
	 *	add name space. long name is the key
	 */
	public final void addNameSpace(Atom url, Atom n)
	{
		current.put(url, n);
	}

    /**
	 * find name space, long name is the key
	 */
	public final Atom findNameSpace(Atom n)
	{
		return (Atom)current.get(n);
	}

    
    public final void push()
    {
        contexts.push(current);
        current = (Hashtable)current.clone();
    }
    
    public final void pop()
    {
        current = (Hashtable)contexts.pop();
    }

	Hashtable current = new Hashtable();

    /**
     * Stack to keep track of contexts
     */
    Stack contexts = new Stack();
}